From 3433e6920a4aaa2177f3503ef08256a58f866b33 Mon Sep 17 00:00:00 2001
From: Wei Huang <wei@redhat.com>
Date: Thu, 17 Jan 2019 17:33:56 +0000
Subject: [PATCH 6/7] aarch64: Set virt-rhel8.0.0 max_cpus to 512

RH-Author: Wei Huang <wei@redhat.com>
Message-id: <20190117173357.31514-3-wei@redhat.com>
Patchwork-id: 84038
O-Subject: [RHEL8 qemu-kvm PATCH v3 2/3] aarch64: Set virt-rhel8.0.0 max_cpus to 512
Bugzilla: 1656504
RH-Acked-by: Andrew Jones <drjones@redhat.com>
RH-Acked-by: Auger Eric <eric.auger@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

This patch increases max_cpus of virt-rhel8.0.0 from 255 to 512.
This decision is made due to a recent trend of ARM servers Red Hat
received. Currently Red Hat has HPE Apollo machines that contains 256
cpu cores. It is reasonable to expect that we will see new machines
with > 256 cores very soon. We have verified this patch partially with a
256-vcpu VM on Apollo and expect 512 vcpus will work as well.

Signed-off-by: Wei Huang <wei@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/arm/virt.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 156721a..21965e4 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -2063,11 +2063,11 @@ static void rhel_machine_class_init(ObjectClass *oc, void *data)
 
     mc->family = "virt-rhel-Z";
     mc->init = machvirt_init;
-    /* Start max_cpus at the maximum QEMU supports. We'll further restrict
-     * it later in machvirt_init, where we have more information about the
+    /* Start with max_cpus set to 512, which is the maximum supported by KVM.
+     * The value may be reduced later when we have more information about the
      * configuration of the particular instance.
      */
-    mc->max_cpus = 255;
+    mc->max_cpus = 512;
     mc->block_default_type = IF_VIRTIO;
     mc->no_cdrom = 1;
     mc->pci_allow_0_address = true;
-- 
1.8.3.1

